app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -515,23 +509,25 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun setHintOpenedAndUnRevealed(isHintUnrevealed: Boolean) {
     viewModel.setHintOpenedAndUnRevealedVisibility(isHintUnrevealed)
-    if (isHintUnrevealed) {
-      val hintBulbAnimation = AnimationUtils.loadAnimation(
-        context,
-        R.anim.hint_bulb_animation
-      ).also { it.interpolator = BounceUpAndDownInterpolator() }
-
-      // The bulb should start bouncing every 30 seconds. Note that an initial delay is used for
-      // cases like configuration changes, or returning from a saved checkpoint.
-      lifecycleSafeTimerFactory.run {
-        activity.runPeriodically(delayMillis = 5_000, periodMillis = 30_000) {
-          return@runPeriodically viewModel.isHintOpenedAndUnRevealed.get()!!.also { playAnim ->
-            if (playAnim) binding.hintBulb.startAnimation(hintBulbAnimation)
+    if (!isConfigChangeStateRetentionEnabled.value) {
```

Comment:
I don't think this is how we want to gate this. This would disable the animation in production code if change stat retention is enabled. We want both to be able to be enabled for users, so this needs to be gated by a separate, new parameter, so that we can specifically disable it in tests while also keeping the config state retention enabled.

Reply:
Created a new platform parameter for Hint bulb animation